src/schema/notifications.ts

```diff
@@ -96,8 +98,78 @@ export class ZimbraNotifications {
 		this.handleContactNotifications(notification);
 	};
 
+	/**
+	 * Processes the specified notification items with the processor function passed in batches with timeout, by queueing them in the
+	 * JavaScript event loop. To prevent the freezing in UI.
+	 *
+	 * @private
+	 * @param {Array<any>} items array of notification items that need to be batched
+	 * @param {Function} processorFn function which processes the items
+	 * @returns
+	 * @memberof ZimbraNotifications
+	 */
+	private batchProcessItems(items: Array<any>, processorFn: Function) {
+		if (!items || (items && items.length === 0)) {
+			return;
+		}
+
+		const BATCH_SIZE = 50;
+		const LENGTH = items.length;
+		const TIMEOUT = 100;
+		const ITERATIONS = Math.ceil(LENGTH / BATCH_SIZE);
+
+		let i: number;
+		for (i = 0; i < ITERATIONS; i++) {
+			const start = i * BATCH_SIZE;
+			let end = start + BATCH_SIZE;
+
+			if (end > LENGTH) {
+				end = LENGTH;
+			}
+
+			const batch = items.slice(start, end);
+
+			// When the timed out function executes, the variables accessed inside the function have to be available through closure
+			// Otherwise, the latest values of the variables would be used, which can have been updated by the loop iterations that executed
+			// after the timeout was set and before it was executed.
+			setTimeout(
+				((i, ITERATIONS, batch) => () => {
+					processorFn(batch);
+					// broadcast updates in the last iteration
+					if (i === ITERATIONS - 1) {
+						this.broadcastCacheUpdates();
+					}
+				})(i, ITERATIONS, batch),
+				TIMEOUT
+			);
+		}
+	}
+
+	private broadcastCacheUpdates = () => {
+		this.getApolloClient().queryManager.broadcastQueries();
```

Comment:
Does calling `writeFragment` already do this?

Reply:
Here I think problem is with cache, we perform `writeFragment` on cache object and which is not broadcasting updated data instantaneously,  https://www.apollographql.com/docs/react/advanced/caching/#direct-cache-access